components/api_request_helper/api_request_helper.h

```diff
@@ -124,17 +131,35 @@ class APIRequestHelper {
     void OnRetry(base::OnceClosure start_retry) override;
 
     // This is used for one shot responses
-    void OnResponse(ResultCallback callback,
-                    ResponseConversionCallback conversion_callback,
+    void OnResponse(ResponseConversionCallback conversion_callback,
                     const std::unique_ptr<std::string> response_body);
 
+    // Decode one shot reponses
+    void OnParseJsonResponse(
+        APIRequestResult result,
+        data_decoder::DataDecoder::ValueOrError result_value);
+
     std::unique_ptr<network::SimpleURLLoader> url_loader_;
     raw_ptr<APIRequestHelper> api_request_helper_;
 
-    // Streaming callbacks
     DataReceivedCallback data_received_callback_;
-    DataCompletedCallback data_completed_callback_;
+    ResultCallback result_callback_;
+    ResponseConversionCallback conversion_callback_;
+
     bool is_sse_ = false;
+
+    // To ensure ordered processing of stream chunks, we create our own
+    // instances of DataDecoder and remote for JsonParser. This avoids the issue
```

Comment:
I don't see that we bind a remote for JsonParser. Perhaps, this comment needs to be updated?

Reply:
cleaned it up!